### PR TITLE
Fixes #2160 population snapshot disease state

### LINKED ISSUE #2160: Population snapshot does not support disease state parmaeters

### PR DESCRIPTION
**Pull request by msevestre**

### CODE REVIEW

File: src/PKSim.Core/Snapshots/Mappers/RandomPopulationSettingsMapper.cs [hunk 1]
```diff
@@ -29,25 +30,47 @@ public class RandomPopulationSettingsMapper : SnapshotMapperBase<RandomPopulatio
...
+         var individual = randomPopulationSettings.BaseIndividual;
+         snapshot.Individual = await _individualMapper.MapToSnapshot(individual);
          snapshot.NumberOfIndividuals = randomPopulationSettings.NumberOfIndividuals;
          snapshot.ProportionOfFemales = proportionOfFemalesFrom(randomPopulationSettings);
          snapshot.Age = await snapshotRangeFor(randomPopulationSettings, CoreConstants.Parameters.AGE);
          snapshot.Weight = await snapshotRangeFor(randomPopulationSettings, CoreConstants.Parameters.MEAN_WEIGHT);
          snapshot.Height = await snapshotRangeFor(randomPopulationSettings, CoreConstants.Parameters.MEAN_HEIGHT);
          snapshot.GestationalAge = await snapshotRangeFor(randomPopulationSettings, Constants.Parameters.GESTATIONAL_AGE);
          snapshot.BMI = await snapshotRangeFor(randomPopulationSettings, CoreConstants.Parameters.BMI);
+         snapshot.DiseaseStateParameters = await snapshotDiseaseStateParameters(individual, randomPopulationSettings);
```
Comment (msevestre) [inline src/PKSim.Core/Snapshots/Mappers/RandomPopulationSettingsMapper.cs:42]: @Yuri05 This was completely missing.

File: src/PKSim.Core/Snapshots/Mappers/RandomPopulationSettingsMapper.cs [hunk 2]
```diff
@@ -61,9 +84,43 @@ public override async Task<RandomPopulationSettings> MapToModel(PopulationSettin
...
+         //Can only be null if someone messes around with snapshots or in tests
+         var diseaseState = randomPopulationSettings.BaseIndividual?.OriginData?.DiseaseState;
+         if (diseaseState == null)
+            return;
+
+         foreach (var snapshotDiseaseStateParameter in snapshot.DiseaseStateParameters)
+         {
+            var diseaseStateParameter = diseaseState.Parameter(snapshotDiseaseStateParameter.Name);
+            if (diseaseStateParameter == null)
+               continue;
```
Comment (Yuri05) [inline src/PKSim.Core/Snapshots/Mappers/RandomPopulationSettingsMapper.cs:106]: shouldn't we show a warning here? "Snapshot disiease state parameter XY not found in ..."

Comment (msevestre) [reply src/PKSim.Core/Snapshots/Mappers/RandomPopulationSettingsMapper.cs]: Ok

Comment (msevestre) [reply src/PKSim.Core/Snapshots/Mappers/RandomPopulationSettingsMapper.cs]: Done

**Review by Yuri05 — APPROVED** [review]

**Review by Yuri05 — APPROVED** [review]

**Pull request merged**